<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Wei Li, Xiaolong Cheng, Lin Yang" />


<title>scMAGeCK</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">scMAGeCK</h1>
<h4 class="author">Wei Li, Xiaolong Cheng, Lin Yang</h4>
<h4 class="date">Jan 6, 2022</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>scMAGeCK is a computational model to identify genes associated with multiple expression phenotypes from CRISPR screening coupled with single-cell RNA sequencing data (e.g., CROP-seq).</p>
<p>scMAGeCK is based on our previous <a href="http://mageck.sourceforge.net">MAGeCK</a> and MAGeCK-VISPR models for pooled CRISPR screens, but further extends to scRNA-seq as the readout of the screening experiment. scMAGeCK consists of two modules: scMAGeCK-Robust Rank Aggregation (RRA), a sensitive and precise algorithm to detect genes whose perturbation links to one single marker expression; and scMAGeCK-LR, a linear-regression based approach that unravels the perturbation effects on thousands of gene expressions, especially from cells undergo multiple perturbations.</p>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<div id="scmageck_rra" class="section level3">
<h3>scmageck_rra</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(scMAGeCK)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="do">### BARCODE file contains cell identity information, generated from the cell identity collection step</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    BARCODE <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;barcode_rec.txt&quot;</span>,<span class="at">package =</span> <span class="st">&quot;scMAGeCK&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="do">### RDS can be a Seurat object or local RDS file path that contains the scRNA-seq dataset</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    RDS <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;singles_dox_mki67_v3.RDS&quot;</span>,<span class="at">package =</span> <span class="st">&quot;scMAGeCK&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Set RRA executable file path. </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="do">###    You can generate RRA executable file by following commands:</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="do">###        wget https://bitbucket.org/weililab/scmageck/downloads/RRA_0.5.9.zip</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="do">###        unzip RRA_0.5.9.zip</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="do">###        cd RRA_0.5.9</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    <span class="do">###        make</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    RRAPATH <span class="ot">&lt;-</span> <span class="st">&quot;/Library/RRA_0.5.9/bin/RRA&quot;</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>    target_gene <span class="ot">&lt;-</span> <span class="st">&quot;MKI67&quot;</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    rra_result <span class="ot">&lt;-</span> <span class="fu">scmageck_rra</span>(<span class="at">BARCODE=</span>BARCODE, <span class="at">RDS=</span>RDS, <span class="at">GENE=</span>target_gene, <span class="at">RRAPATH=</span>RRAPATH, </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">LABEL=</span><span class="st">&#39;dox_mki67&#39;</span>, <span class="at">NEGCTRL=</span><span class="cn">NULL</span>, <span class="at">KEEPTMP=</span><span class="cn">FALSE</span>, <span class="at">PATHWAY=</span><span class="cn">FALSE</span>, <span class="at">SAVEPATH=</span><span class="cn">NULL</span>)</span></code></pre></div>
<pre><code>## Testing Rcpp:</code></pre>
<pre><code>## Checking RRA...</code></pre>
<pre><code>## Warning in system(&quot;RRA&quot;, ignore.stdout = TRUE, ignore.stderr = TRUE): error in
## running command</code></pre>
<pre><code>## RRA does not exist! Please check RRA executable file path</code></pre>
<pre><code>## keep_tmp: FALSE</code></pre>
<pre><code>## ispathway: FALSE</code></pre>
<pre><code>## Only assign unique cells</code></pre>
<pre><code>## Total barcode records: 8425</code></pre>
<pre><code>## Unique barcode records: 5282</code></pre>
<pre><code>## Reading RDS file: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scMAGeCK/extdata/singles_dox_mki67_v3.RDS</code></pre>
<pre><code>## Cell names in expression matrix and barcode file do not match. Try to remove possible trailing &quot;-1&quot;s...</code></pre>
<pre><code>## Target gene: MKI67</code></pre>
<pre><code>## Testing gene  MKI67 ...</code></pre>
<pre><code>## Calling RRA parameters:
## /Library/RRA_0.5.9/bin/RRA -i sample_1568.33024852839_rra_low.txt -o sample_1568.33024852839_rra_low.out  -p 0.3 --max-sgrnapergene-permutation 10000  
## Calling RRA parameters:
## /Library/RRA_0.5.9/bin/RRA -i sample_1568.33024852839_rra_high.txt -o sample_1568.33024852839_rra_high.out  -p 0.3 --max-sgrnapergene-permutation 10000</code></pre>
</div>
<div id="scmageck_lr" class="section level3">
<h3>scmageck_lr</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(scMAGeCK)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    <span class="do">### BARCODE file contains cell identity information, generated from the cell identity collection step</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    BARCODE <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;barcode_rec.txt&quot;</span>,<span class="at">package =</span> <span class="st">&quot;scMAGeCK&quot;</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    <span class="do">### RDS can be a Seurat object or local RDS file path that contains the scRNA-seq dataset</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    RDS <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;singles_dox_mki67_v3.RDS&quot;</span>,<span class="at">package =</span> <span class="st">&quot;scMAGeCK&quot;</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    lr_result <span class="ot">&lt;-</span> <span class="fu">scmageck_lr</span>(<span class="at">BARCODE=</span>BARCODE, <span class="at">RDS=</span>RDS, <span class="at">LABEL=</span><span class="st">&#39;dox_scmageck_lr&#39;</span>, </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">NEGCTRL =</span> <span class="st">&#39;NonTargetingControlGuideForHuman&#39;</span>, <span class="at">PERMUTATION =</span> <span class="dv">1000</span>, <span class="at">SAVEPATH=</span><span class="cn">NULL</span>, <span class="at">LAMBDA=</span><span class="fl">0.01</span>)</span></code></pre></div>
<pre><code>## run_signature: FALSE</code></pre>
<pre><code>## Total barcode records: 8425</code></pre>
<pre><code>## Neg Ctrl guide: NonTargetingControlGuideForHuman</code></pre>
<pre><code>## Reading RDS file: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/scMAGeCK/extdata/singles_dox_mki67_v3.RDS</code></pre>
<pre><code>## Cell names in expression matrix and barcode file do not match. Try to remove possible trailing &quot;-1&quot;s...</code></pre>
<pre><code>## 6704 ...</code></pre>
<pre><code>## 6229 ...</code></pre>
<pre><code>## Index matrix dimension: 5698 , 30</code></pre>
<pre><code>## Filter genes whose expression is greater than 0 in raw read count in less than 0.01 single-cell populations.</code></pre>
<pre><code>## Selected genes: 25</code></pre>
<pre><code>## Selected cells: 5698</code></pre>
<pre><code>## Permutation: 100 / 1000 ...</code></pre>
<pre><code>## Permutation: 200 / 1000 ...</code></pre>
<pre><code>## Permutation: 300 / 1000 ...</code></pre>
<pre><code>## Permutation: 400 / 1000 ...</code></pre>
<pre><code>## Permutation: 500 / 1000 ...</code></pre>
<pre><code>## Permutation: 600 / 1000 ...</code></pre>
<pre><code>## Permutation: 700 / 1000 ...</code></pre>
<pre><code>## Permutation: 800 / 1000 ...</code></pre>
<pre><code>## Permutation: 900 / 1000 ...</code></pre>
<pre><code>## Permutation: 1000 / 1000 ...</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>    lr_score <span class="ot">&lt;-</span> lr_result[<span class="dv">1</span>]</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    lr_score_pval <span class="ot">&lt;-</span> lr_result[<span class="dv">2</span>]</span></code></pre></div>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<div id="scmageck_rra-1" class="section level3">
<h3>scmageck_rra</h3>
<p>The scmageck_rra function will output the ranking and p values of each perturbed genes, using the RRA program in MAGeCK. Users familiar with the MAGeCK program may find it similar with the <a href="https://sourceforge.net/p/mageck/wiki/output/#gene_summary_txt">gene_summary</a> output in MAGeCK.</p>
<p>Here is the example output of scMAGeCK-RRA:</p>
<pre><code>Row.names  items_in_group.low  lo_value.low  p.low  FDR.low goodsgrna.low  items_in_group.high  lo_value.high  p.high  FDR.high  goodsgrna.high
TP53    271     0.11832 0.95619 1       48      271     1.014e-83       4.9975e-06      0.00015 184</code></pre>
<p>Explanations of each column are below:</p>
<table>
<colgroup>
<col width="46%" />
<col width="53%" />
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Row.names</td>
<td>Perturbed gene name</td>
</tr>
<tr class="even">
<td>items_in_group.low</td>
<td>The number of single-cells with each gene perturbed</td>
</tr>
<tr class="odd">
<td>lo_value.low</td>
<td>The RRA score in negative selection (reducing the marker expression if this gene is perturbed). The RRA score uses a p value from rank order statistics to measure the degree of selection; the smaller score, the stronger the selection is. More information on the calculation of RRA score can be found in our original <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0554-4">MAGeCK</a> paper.</td>
</tr>
<tr class="even">
<td>p.low</td>
<td>The raw p-value (using permutation) of this gene in negative selection</td>
</tr>
<tr class="odd">
<td>FDR.low</td>
<td>The false discovery rate of this gene in negative selection</td>
</tr>
<tr class="even">
<td>goodsgrna.low</td>
<td>The number of single-cells that passes the threshold and is considered in the RRA score calculation in negative selection</td>
</tr>
<tr class="odd">
<td>items_in_group.high</td>
<td>The same as items_in_group.low: the number of single-cells with each gene perturbed)</td>
</tr>
<tr class="even">
<td>lo_value.high</td>
<td>The RRA score in positive selection (increasing the marker expression if this gene is perturbed</td>
</tr>
<tr class="odd">
<td>p.high</td>
<td>The raw p-value (using permutation) of this gene in positive selection</td>
</tr>
<tr class="even">
<td>FDR.high</td>
<td>The false discovery rate of this gene in positive selection</td>
</tr>
<tr class="odd">
<td>goodsgrna.high</td>
<td>The number of single-cells that passes the threshold and is considered in the RRA score calculation in positive selection</td>
</tr>
</tbody>
</table>
</div>
<div id="scmageck_lr-1" class="section level3">
<h3>scmageck_lr</h3>
<p>The scmageck_lr function will generate several files below:</p>
<table>
<colgroup>
<col width="28%" />
<col width="71%" />
</colgroup>
<thead>
<tr class="header">
<th>File</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>lr_score</td>
<td>The score (similar with log fold change) of each perturbed gene (rows) on each marker gene (columns)</td>
</tr>
<tr class="even">
<td>lr_score.pval</td>
<td>The associated p values of each score</td>
</tr>
<tr class="odd">
<td>LR.RData</td>
<td>An R object to store scores and p values</td>
</tr>
</tbody>
</table>
<p>The format of score.txt and score.pval.txt is a simple table file with rows corresponding to perturbed genes and columns corresponding to marker genes. For example in the score.txt,</p>
<pre><code>Perturbedgene  APC                ARID1A               TP53               MKI67
     APC       0.138075836476524  -0.0343441660045313  0.214449590551132  -0.150287676553705</code></pre>
<p>This row records the effects of perturbing APC gene on the expressions of APC, ARID1A, TP53 and MKI67.</p>
</div>
</div>
<div id="contact-us" class="section level2">
<h2>Contact us</h2>
<p>Questions? Comments? Join the <a href="https://groups.google.com/forum/?hl=en#!forum/mageck">MAGeCK Google group</a> or email us (<a href="mailto:wli2@childrensnational.org" class="email">wli2@childrensnational.org</a>) directly.</p>
<p>Any advice and suggestions will be greatly appreciated.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
